lib/ansible/modules/system/aix_lvol.py

```diff
@@ -0,0 +1,354 @@
+            policy=dict(choices=["maximum", "minimum"], default='maximum'),
+            pvs=dict(type='list',default=list())
+        ),
+        supports_check_mode=True,
+    )
+
+    vg = module.params['vg']
+    lv = module.params['lv']
+    lv_type = module.params['lv_type']
+    size = module.params['size']
+    opts = module.params['opts']
+    copies = module.params['copies']
+    policy = module.params['policy']
+    state = module.params['state']
+    pvs = module.params['pvs']
+
+    pv_list = ' '.join(pvs)
+
+    if policy == "maximum":
+        lv_policy="x"
+    else:
+        lv_policy="m"
+
+    # Add echo command when running in check-mode
+    if module.check_mode:
+        test_opt = 'echo '
+    else:
+        test_opt = ''
+
+    # check if system commands are available
+    lsvg_cmd = module.get_bin_path("lsvg", required=True)
+    lslv_cmd = module.get_bin_path("lslv", required=True)
+
+    # Get information on volume group requested
+    rc, vg_info, err = module.run_command("%s %s" % (lsvg_cmd, vg))
+
+    if rc != 0:
+        if state == 'absent':
+            module.exit_json(changed=False, msg="Volume group %s does not exist." % vg)
+        else:
+            module.fail_json(msg="Volume group %s does not exist." % vg, rc=rc, err=err)
+
+    this_vg = parse_vg(vg_info)
+
+
+    if size is not None:
+        # Calculate pp size and round it up based on pp size.
+        lv_size = round_ppsize(convert_size(size), base=this_vg['pp_size'])
+
+    # Get information on logical volume requested
+    rc, lv_info, err = module.run_command(
+        "%s %s" % (lslv_cmd, lv))
+
+    if rc != 0:
+        if state == 'absent':
+            module.exit_json(changed=False, msg="Logical Volume %s does not exist." % lv)
+
+    changed = False
+
+    this_lv = parse_lv(lv_info)
+
+    if state == 'present' and not size:
+        if this_lv is None:
+            module.fail_json(msg="No size given.")
+
+    if this_lv is None:
+        if state == 'present':
+            if lv_size > this_vg['free']:
+                module.fail_json(msg="Not enough free space in volume group %s: %s MB free." % (this_vg['name'], this_vg['free']))
+
+            ### create LV
+            mklv_cmd = module.get_bin_path("mklv", required=True)
+
+            cmd = "%s %s -t %s -y %s -c %s  -e %s %s %s %sM %s" % (test_opt, mklv_cmd, lv_type, lv, copies, lv_policy, opts, vg, lv_size, pv_list)
+            rc, _, err = module.run_command(cmd)
+            if rc == 0:
+                module.exit_json(changed=True, msg="Logical volume %s created." % lv)
+            else:
+                module.fail_json(msg="Creating logical volume '%s' failed" % lv, rc=rc, err=err)
+    else:
+        if state == 'absent':
+            ### remove LV
+            rmlv_cmd = module.get_bin_path("rmlv", required=True)
+            rc, _, err = module.run_command("%s %s -f %s" % (test_opt, rmlv_cmd, this_lv['name']))
```

Comment:
We are not doing anything with stdout and stderr, but on failures it may be the only clue to why it failed.